src/main/java/mate/academy/springboot/swagger/controller/ProductController.java

```diff
@@ -0,0 +1,114 @@
+package mate.academy.springboot.swagger.controller;
+
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.Parameter;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import mate.academy.springboot.swagger.dto.request.ProductRequestDto;
+import mate.academy.springboot.swagger.dto.response.ProductResponseDto;
+import mate.academy.springboot.swagger.model.Product;
+import mate.academy.springboot.swagger.service.ProductService;
+import mate.academy.springboot.swagger.service.SortParserService;
+import mate.academy.springboot.swagger.service.mapper.DtoMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Sort;
+import org.springframework.web.bind.annotation.*;
+
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@RestController
+@Tag(name = "Products API")
+@RequestMapping("/products")
+public class ProductController {
+    private final ProductService productService;
+    private final DtoMapper<Product, ProductResponseDto, ProductRequestDto> dtoDtoMapper;
+    private final SortParserService sortParserService;
+
+    @Autowired
```

Comment:
Autowired annotation on constructor is not necessary. Since Spring 4.3, it automatically applies @Autowired if the target bean only defines one constructor.